hi-ml-multimodal/src/health_multimodal/vlp/inference_engine.py

```diff
@@ -19,23 +19,19 @@
 
 
 class ImageTextInferenceEngine:
-    """
-    Encapsulate functions related to inference on ImageTextModels.
-    """
+    """Functions related to inference on :class:`ImageTextModel`."""
 
     def __init__(self,
                  image_inference_engine: ImageInferenceEngine,
                  text_inference_engine: TextInferenceEngine) -> None:
-        """
-        This class takes an ImageTextModel as well as an ImageInferenceEngine and TextInferenceEngine.
-        """
         self.image_inference_engine = image_inference_engine
         self.text_inference_engine = text_inference_engine
 
-    def get_similarity_map_from_raw_data(self, image_path: Path, query_text: str) -> np.ndarray:
-        """
-        Return a heatmap of the similarities between each patch embedding from the image and the text embedding.
-        """
+    def get_similarity_map_from_raw_data(self,
+                                         image_path: Path,
+                                         query_text: str,
+                                         interpolation: str = "nearest") -> np.ndarray:
+        """Return a heatmap of the similarities between each patch embedding from the image and the text embedding."""
```

Comment:
can we update the documentation here? What kind of image formats are supported, max_text length (e.g., 512), and where this interpolation is actually used?

Reply:
Is there a maximum text length?

Reply:
I've improved the docstring in b8715c8. Let me know if there is a maximum length and I'll add that.